crates/uv-requirements/src/lib.rs

```diff
@@ -40,27 +42,21 @@ pub enum Error {
 impl Error {
     /// Create an [`Error`] from a distribution error.
     pub(crate) fn from_dist(dist: Dist, cause: uv_distribution::Error) -> Self {
-        match dist {
-            Dist::Built(dist) => Self::Dist(
-                DistErrorKind::Download,
-                Box::new(Dist::Built(dist)),
-                DerivationChain::default(),
-                cause,
-            ),
-            Dist::Source(dist) => {
-                let kind = if dist.is_local() {
-                    DistErrorKind::Build
-                } else {
-                    DistErrorKind::DownloadAndBuild
-                };
-                Self::Dist(
-                    kind,
-                    Box::new(Dist::Source(dist)),
-                    DerivationChain::default(),
-                    cause,
-                )
+        let kind = if cause.is_build_backend_error() {
+            DistErrorKind::BuildBackend
+        } else {
+            match &dist {
+                Dist::Built(_) => DistErrorKind::Download,
+                Dist::Source(dist) => {
+                    if dist.is_local() {
+                        DistErrorKind::Build
+                    } else {
+                        DistErrorKind::DownloadAndBuild
+                    }
+                }
             }
-        }
+        };
+        Self::Dist(kind, Box::new(dist), DerivationChain::default(), cause)
```

Comment:
Doesn't need to happen here, but is it feasible for us to consolidate this repeated logic?